File: Trabalho_Final/TrabalhoFinal/src/servidorJS/Despachante.js
const Esqueleto = require('./Esqueleto');

class Despachante {
    esqueleto = new Esqueleto();

    invoke(message) {

        switch (message.objectReference) {
            case "IMC":
                return this.esqueleto.IMC(message.arguments);
            case "Bike":
                return this.esqueleto.Bike(message.arguments);
            case "Health":
                return this.esqueleto.Health(message.arguments);
            default:
                return [ ":(" ];
        }
    }
}

module.exports = Despachante;

File: Trabalho_Final/TrabalhoFinal/src/servidor/Despachante.java
package servidor;

import org.json.JSONObject;

public class Despachante {

    Esqueleto esqueleto = new Esqueleto();

    public String invoke(String dados) {
        JSONObject json = new JSONObject(dados);

        switch (json.getString("calculo")) {
            case "IMC":
                return esqueleto.IMC(json);
            case "alturaBike":
                return esqueleto.alturaBike(json);
            case "batimentosCardiacos":
                return esqueleto.batimentosCardiacos(json);
            default:
                return "";
        }
    }
}
